echo_test.go

```diff
@@ -773,8 +774,23 @@ var listenerNetworkTests = []struct {
 	{"tcp6 ipv6 address", "tcp6", "[::1]:1323"},
 }
 
+func supportsIPv6() bool {
+	addrs, _ := net.InterfaceAddrs()
+	for _, addr := range addrs {
+		// Check if any interface has local IPv6 assigned
+		if strings.Contains(addr.String(), "::1") {
+			return true
+		}
+	}
+	return false
+}
+
 func TestEchoListenerNetwork(t *testing.T) {
 	for _, tt := range listenerNetworkTests {
+		if !supportsIPv6() && strings.Contains(tt.address, "::") {
```

Comment:
Please store the value of supportsIPv6() in a variable before the for to avoid getting Interfaces information for every test

Reply:
nice catch!

Reply:
Oops. Fixing